### PR TITLE
[FLINK-31967][table-runner] Solving the Initial Value Problem of LagA…

### PR DESCRIPTION
**Pull request by whjshj**
…ggregation

## What is the purpose of the change

Solve the problem of initializing default values for LagAggFunction

## Brief change log

  - *Judge whether the default value can be empty*
  - *Added testing for this change*

## Verifying this change

This change added tests and can be verified as follows:
org.apache.flink.table.planner.runtime.stream.sql#testLagAggFunction

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Review by fsk119 — COMMENTED** [review]
Thanks for your contribution. Please also add a case for lead function.

File: flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/aggregate/LagAggFunction.java [hunk 1]
```diff
@@ -44,6 +44,11 @@ public LagAggFunction(LogicalType[] valueTypes) {
                 Arrays.stream(valueTypes)
                         .map(DataTypeUtils::toInternalDataType)
                         .toArray(DataType[]::new);
+        //When the initial value has not been initialized and is not allowed to be null,
+        // it should be changed to allow nulls
+        if (valueDataTypes.length < 3 && !valueDataTypes[0].getLogicalType().isNullable()) {
+            valueDataTypes[0] = valueDataTypes[0].nullable();
+        }
```
[the remark this answers is not in the document]
Comment (whjshj) [reply flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/aggregate/LagAggFunction.java:51]: yes.if user specify the lag(a,1,null),the length of valueDataTypes.length == 3,so the type of the first argument would not be changed.

File: flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/aggregate/LagAggFunction.java [hunk 2]
```diff
@@ -44,6 +44,15 @@ public LagAggFunction(LogicalType[] valueTypes) {
                 Arrays.stream(valueTypes)
                         .map(DataTypeUtils::toInternalDataType)
                         .toArray(DataType[]::new);
+        // When the initial value has not been initialized and is not allowed to be null or
+        // default value allowed to be empty,output should be changed to allow nulls
```
Comment (fsk119) [inline flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/aggregate/LagAggFunction.java:48]: The output value can only be not null if the default input arguments include a non-null default value.

File: flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala [hunk 3]
```diff
@@ -1765,4 +1765,35 @@ class AggregateITCase(aggMode: AggMode, miniBatch: MiniBatchMode, backend: State
...
+
+  @Test
+  def testLagAggFunction(): Unit = {
+    val data =
+      List(rowOf(2L, 15, "Hello"), rowOf(8L, 11, "Hello world"), rowOf(9L, 12, "Hello world!"))
+    val dataId = TestValuesTableFactory.registerData(data)
+    tEnv.executeSql(s"""
+                       |CREATE TABLE src(
+                       |  `id` BIGINT,
+                       |  len INT,
```
Comment (fsk119) [inline flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala:1777]: ditto

**Review by fsk119 — COMMENTED** [review]
Thanks for your update. I left some minor suggestions.

**Review by fsk119 — COMMENTED** [review]
Thanks for your update. I left some minor suggestions.

**Review by fsk119 — APPROVED** [review]
LGTM

**Pull request merged**